### PR TITLE
Functionality to tune routerWorkers per destination type

### PR DESCRIPTION
**Pull request by bipin-rudder**
**Fixes** # (*issue*)

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: router/batchrouter/batchrouter.go [hunk 1]
```diff
@@ -831,6 +830,7 @@ func (brt *HandleT) Setup(jobsDB *jobsdb.HandleT, destType string) {
 	brt.destType = destType
 	brt.jobsDB = jobsDB
 	brt.isEnabled = true
+	brt.noOfWorkers = getRouterConfigInt("noOfWorkers", destType, 8)
```
Comment (chandumlg) [inline router/batchrouter/batchrouter.go:833]: rename this to getBatchRouterConfigInt

File: router/router.go [hunk 2]
```diff
@@ -585,8 +585,9 @@ func (rt *HandleT) trackRequestMetrics(reqMetric requestMetric) {
 }
 
 func (rt *HandleT) initWorkers() {
-	rt.workers = make([]*workerT, noOfWorkers)
-	for i := 0; i < noOfWorkers; i++ {
+	rt.workers = make([]*workerT, rt.noOfWorkers)
+	for i := 0; i < rt.noOfWorkers; i++ {
+		logger.Info("Worker Started", i)
```
Comment (chandumlg) [inline router/router.go:590]: Remove this log.

File: router/batchrouter/batchrouter.go [hunk 3]
```diff
@@ -455,7 +455,7 @@ func (brt *HandleT) recordDeliveryStatus(batchDestination DestinationT, err erro
 }
 
 func (brt *HandleT) initWorkers() {
-	for i := 0; i < noOfWorkers; i++ {
+	for i := 0; i < brt.noOfWorkers; i++ {
```
Comment (chandumlg) [inline router/batchrouter/batchrouter.go:458]: One more thing, not related to this PR. 

initWorkers need not be called inside a goroutine. Not a problem, but unncessary.

Can you check with @psrikanth88 ?

File: router/router.go (hunk 2 above)
Comment (bipin-rudder) [reply router/router.go]: done

File: router/batchrouter/batchrouter.go (hunk 1 above)
Comment (bipin-rudder) [reply router/batchrouter/batchrouter.go]: done

**Review by chandumlg — APPROVED** [review]

**Review by snarkychef — APPROVED** [review]

**Pull request merged**